homeassistant/components/bmp280/sensor.py

```diff
@@ -0,0 +1,161 @@
+"""Platform for Bosch BMP280 Environmental Sensor integration."""
+import logging
+
+from adafruit_bmp280 import Adafruit_BMP280_I2C
+from busio import I2C
+import voluptuous as vol
+
+from homeassistant.components.sensor import (
+    DEVICE_CLASS_PRESSURE,
+    DEVICE_CLASS_TEMPERATURE,
+    PLATFORM_SCHEMA,
+)
+from homeassistant.const import CONF_NAME, PRESSURE_HPA, TEMP_CELSIUS
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "BMP280"
+DEFAULT_I2C_ADDRESS = 0x77
+
+MIN_I2C_ADDRESS = 0x76
+MAX_I2C_ADDRESS = 0x77
+
+CONF_I2C_ADDRESS = "i2c_address"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_I2C_ADDRESS, default=DEFAULT_I2C_ADDRESS): vol.All(
+            vol.Coerce(int), vol.Range(min=MIN_I2C_ADDRESS, max=MAX_I2C_ADDRESS)
+        ),
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the sensor platform."""
+    try:
+        # this throws an exception if board is not supported
+        import board
+
+        # initializing I2C bus using the auto-detected pins
+        i2c = I2C(board.SCL, board.SDA)
+        # initializing the sensor
+        bmp280 = Adafruit_BMP280_I2C(i2c, address=config.get(CONF_I2C_ADDRESS))
+        # use custom name if there's any
+        name = config.get(CONF_NAME)
+        # BMP280 has both temperature and pressure sensing capability
+        add_entities(
+            [Bmp280TemperatureSensor(bmp280, name), Bmp280PressureSensor(bmp280, name)]
+        )
+    except NotImplementedError as error:
+        # this is thrown right after the import statement if the board is unsupported
+        if error.args[0] == "Board not supported":
+            _LOGGER.error(
+                "Failed to determine board type. Is this instance running on a Raspberry Pi or a similar I2C capable device?"
+            )
+            raise PlatformNotReady()
```

Comment:
This doesn't seem like its recoverable.

Please change to return to prevent `helpers/entity_platform.py` from trying to set it up again

Reply:
You mean it should swallow the exception and return from the function normally or should it rethrow the exception after logging?

Reply:
entity_platform._async_setup_platform is going to log any exception already so you could likely just remove check for the NotImplementedError exception. 

```
       except Exception:  # pylint: disable=broad-except
            logger.exception(
                "Error while setting up %s platform for %s",
                self.platform_name,
                self.domain,
            )
            return False
```